kube-core/src/conversion/types.rs

```diff
@@ -0,0 +1,178 @@
+use crate::{Status, TypeMeta};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// The `kind` field in [`TypeMeta`].
+pub const META_KIND: &str = "ConversionReview";
+/// The `api_version` field in [`TypeMeta`] on the v1 version.
+pub const META_API_VERSION_V1: &str = "apiextensions.k8s.io/v1";
+
+#[derive(Debug, Error)]
+#[error("request missing in ConversionReview")]
+/// Failed to convert `AdmissionReview` into `AdmissionRequest`.
+pub struct ConvertConversionReviewError;
+
+/// Struct that describes both request and response.
+#[derive(Serialize, Deserialize)]
+pub struct ConversionReview {
+    /// Contains the API version and type of the request.
+    #[serde(flatten)]
+    pub types: TypeMeta,
+    /// Contains conversion request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub request: Option<ConversionRequest>,
+    /// Contains conversion response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub response: Option<ConversionResponse>,
+}
+
+/// Part of ConversionReview which is set on input (i.e. generated by apiserver)
+#[derive(Serialize, Deserialize)]
+pub struct ConversionRequest {
+    /// Copied from the corresponding consructing [`ConversionReview`].
+    /// This field is not part of the Kubernetes API, it's consumed only by `kube`.
+    #[serde(skip)]
+    pub types: Option<TypeMeta>,
```

Comment:
No strong reason. It simplifies manual creation of `ConversionRequest` a little bit (i.e. it may be surprising for user to see non-optional field which is not set by apiserver).